earn/src/components/advanced/BorrowMetrics.tsx

```diff
@@ -166,62 +172,30 @@ export type BorrowMetricsProps = {
-      marginAccount.assets,
-      marginAccount.liabilities,
-      marginAccount.assets.uniswapPositions,
-      marginAccount.sqrtPriceX96,
-      marginAccount.iv,
-      marginAccount.nSigma,
-      marginAccount.token0.decimals,
-      marginAccount.token1.decimals
-    );
-
-    if (lowerSqrtRatio.eq(minSqrtRatio) && upperSqrtRatio.eq(maxSqrtRatio)) return Number.POSITIVE_INFINITY;
-
-    const [current, lower, upper] = [marginAccount.sqrtPriceX96, lowerSqrtRatio, upperSqrtRatio].map((sp) =>
-      sqrtRatioToPrice(sp, marginAccount.token0.decimals, marginAccount.token1.decimals)
-    );
-
-    const assets = getAssets(marginAccount.assets, lowerSqrtRatio, upperSqrtRatio);
-
-    // Compute the value of all assets (collateral) at 3 different prices (current, lower, and upper)
-    // Denominated in units of token1
-    let assetValueCurrent = token0Collateral * current + token1Collateral;
-    let assetValueAtLower = assets.amount0AtA * lower + assets.amount1AtA;
-    let assetValueAtUpper = assets.amount0AtB * upper + assets.amount1AtB;
-
-    // If there are no assets, further results would be spurious, so return null
-    if (assetValueCurrent < Number.EPSILON) return null;
-
-    // Compute how much the collateral can drop in value while remaining solvent
-    const percentChange1A = Math.abs(assetValueCurrent - assetValueAtLower) / assetValueCurrent;
-    const percentChange1B = Math.abs(assetValueCurrent - assetValueAtUpper) / assetValueCurrent;
-    const percentChange1 = Math.min(percentChange1A, percentChange1B);
-
-    // Now change to units of token0
-    assetValueCurrent /= current;
-    assetValueAtLower /= lower;
-    assetValueAtUpper /= upper;
-
-    // Again compute how much the collateral can drop in value while remaining solvent,
-    // but this time percentages are based on units of token0
-    const percentChange0A = Math.abs(assetValueCurrent - assetValueAtLower) / assetValueCurrent;
-    const percentChange0B = Math.abs(assetValueCurrent - assetValueAtUpper) / assetValueCurrent;
-    const percentChange0 = Math.min(percentChange0A, percentChange0B);
-
-    // Since we don't know whether the user is thinking in terms of "X per Y" or "Y per X",
-    // we return the minimum. Error on the side of being too conservative.
-    return Math.min(percentChange0, percentChange1);
-  }, [marginAccount, token0Collateral, token1Collateral]);
+  useEffect(() => {
+    (async () => {
+      setMostRecentModifyTime(null);
+      if (!marginAccount?.mostRecentModify) return;
+      const block = await marginAccount.mostRecentModify.getBlock();
+      setMostRecentModifyTime(new Date(block.timestamp * 1000));
+    })();
+  }, [marginAccount, setMostRecentModifyTime]);
+
+  useEffect(() => {
+    const interval = setInterval(() => setCurrentTime(Date.now()), 200);
+    if (!marginAccount?.warningTime) clearInterval(interval);
+    return () => clearInterval(interval);
+  }, [marginAccount?.warningTime]);
```

Comment:
If you wrap this entire block in the if statement (including the return), it would be more efficient as it would run if the condition is met.

Reply:
But if the `marginAccount` was previous defined, but now it's not, I still want to clear the interval timer.